### PR TITLE
Review

### PR DESCRIPTION
**Pull request by Adanel**

### CODE REVIEW

File: MotorizationRate.ipynb [hunk 1]
```diff
@@ -148,7 +148,7 @@
     }
```
Comment (GuillaumeDesforges) [inline MotorizationRate.ipynb:148]: <p>Python convention: write variables in <code>snake_case</code>  which is lowercase separaed by underscores.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/Adanel/MotorizationRate/pull/3/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='3'/>

Comment (GuillaumeDesforges) [inline MotorizationRate.ipynb:148]: <p>It can be useful to define</p><p> </p><pre class="ql-syntax" spellcheck="false">target = "Taux de motorisation Commune"
</pre><p> </p><p>and then use <code>taget</code>  instead in the rest of the code.</p><p> </p><ul><li>It explicits the target in the code</li><li>It explicits when we are talking about the target in the following code</li><li>It makes the following code more independant of the input data (thus it becomes reusable, or can adapt more easily)</li></ul>
<br/>

 _Reply via <a href='https://app.reviewnb.com/Adanel/MotorizationRate/pull/3/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='5'/>

Comment (GuillaumeDesforges) [inline MotorizationRate.ipynb:148]: <pre class="ql-syntax" spellcheck="false">print('\n'.join(map(str, df_mdb.columns)))
</pre><p>Can be more agreable to read</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/Adanel/MotorizationRate/pull/3/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='8'/>

Comment (GuillaumeDesforges) [inline MotorizationRate.ipynb:148]: <p>Comment what CODGEO is, why do you count nunique ?</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/Adanel/MotorizationRate/pull/3/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='13'/>

Comment (GuillaumeDesforges) [inline MotorizationRate.ipynb:148]: <p>missing title and axis labels</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/Adanel/MotorizationRate/pull/3/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='14'/>

Comment (GuillaumeDesforges) [inline MotorizationRate.ipynb:148]: <p>use markdown titles to structure the notebook</p><p> </p><p>e.g.</p><p> </p><pre class="ql-syntax" spellcheck="false"># Title of the notebook

## Load data

## Exploration

### "Urbanité Ruralité"
</pre><p> </p><p>can even sometime generate TOC</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/Adanel/MotorizationRate/pull/3/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='16'/>

Comment (GuillaumeDesforges) [inline MotorizationRate.ipynb:148]: <p>keep imports at the top</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/Adanel/MotorizationRate/pull/3/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='17'/>

Comment (GuillaumeDesforges) [inline MotorizationRate.ipynb:148]: <p>title, axis labels</p><p> </p><p>what is the definition of the column "Urbanité Ruralié" ?</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/Adanel/MotorizationRate/pull/3/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='18'/>

Comment (GuillaumeDesforges) [inline MotorizationRate.ipynb:148]: <p>probably*, surely*</p><p> </p><p>Issue: <code>.value_counts()</code>  means the counts are on the full content of the "cells": if the last category "contains" have entries that <em>contain</em> other entries, you probably need a group by and make clear categories.</p><p>If the indexing column is a primary key (no duplicate values) then no, the last category does NOT contain others.</p><p> </p><p>Note that it is bad practive to designate "last category" when the code does not ensure order of categories in the chat. If rerunning the notebook leads to another category being the last presented, this cancels your explanation.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/Adanel/MotorizationRate/pull/3/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='19'/>

Comment (GuillaumeDesforges) [inline MotorizationRate.ipynb:148]: <p>title, but good plot</p><p> </p><p>also try swarm plot or violin plot</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/Adanel/MotorizationRate/pull/3/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='20'/>

Comment (GuillaumeDesforges) [inline MotorizationRate.ipynb:148]: <p>YES, a few good features perform better than feeding a lot of garbage.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/Adanel/MotorizationRate/pull/3/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='22'/>

Comment (GuillaumeDesforges) [inline MotorizationRate.ipynb:148]: <p>define FUNCTIONS to make things:</p><p> </p><pre class="ql-syntax" spellcheck="false">def select_categorical_columns(df, numerical_columns=num_cols):
    return [col for col in df.columns if col not in numerical_columns]
</pre><p> </p><p>this is a very good habit to have.</p><p> </p><ul><li>It prevents from having intermediary variables in global scope (hence reduces bugs).</li><li>It allows for easier testing/debugging.</li><li>Reusable pieces</li></ul>
<br/>

 _Reply via <a href='https://app.reviewnb.com/Adanel/MotorizationRate/pull/3/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='27'/>

Comment (GuillaumeDesforges) [inline MotorizationRate.ipynb:148]: <p>bad.</p><p> </p><pre class="ql-syntax" spellcheck="false"># good practice: immutable
df = df.drop(columns="CODGEO")
</pre><p> </p><pre class="ql-syntax" spellcheck="false"># not so good practice but can be useful on BIG dataframes (I mean VERY BIG)
df.drop(columns="CODEGEO", inplace=True)
</pre>
<br/>

 _Reply via <a href='https://app.reviewnb.com/Adanel/MotorizationRate/pull/3/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='29'/>

Comment (GuillaumeDesforges) [inline MotorizationRate.ipynb:148]: <p>Write at least a little comment at start of cell</p><p> </p><p>e.g.</p><p> </p><pre class="ql-syntax" spellcheck="false"># List columns which have at least one NA value
</pre>
<br/>

 _Reply via <a href='https://app.reviewnb.com/Adanel/MotorizationRate/pull/3/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='30'/>

Comment (GuillaumeDesforges) [inline MotorizationRate.ipynb:148]: <p>not PEP8 valid</p><p> </p><p>personal opinion, but comments must not follow a line of code but be on the line above</p><p> </p><pre class="ql-syntax" spellcheck="false">this_is()
a_bad_thing() # do a bad thing
</pre><p> </p><p>vs</p><p> </p><pre class="ql-syntax" spellcheck="false">this_is()
# make it right
a_good_thing()
</pre><p> </p><p>this enhances readability</p><p> </p><p>also, <code>cor_selector</code> is not an understandable name, so you need to put a comment</p><p> </p><p>also, VERY BAD hard coded global <code>num_feats</code> <span class="ql-cursor">﻿</span></p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/Adanel/MotorizationRate/pull/3/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='32'/>

Comment (GuillaumeDesforges) [inline MotorizationRate.ipynb:148]: <p>Validate that the target has a linear relationship with those features using plots.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/Adanel/MotorizationRate/pull/3/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='35'/>

Comment (GuillaumeDesforges) [inline MotorizationRate.ipynb:148]: <p>Try to plot map { x: feature, y: target } for those selected features with the predictor regions (/heatmap), for instance using gridmesh, to see if it is meaningful</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/Adanel/MotorizationRate/pull/3/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='37'/>

Comment (GuillaumeDesforges) [inline MotorizationRate.ipynb:148]: <p>Be careful about feature selection. It would require one level more of train-validate-test split, or crossvalidation, since it is a hyperparameter search.</p><p>You might be overfitting.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/Adanel/MotorizationRate/pull/3/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='39'/>

Comment (GuillaumeDesforges) [inline MotorizationRate.ipynb:148]: <p>big space ???</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/Adanel/MotorizationRate/pull/3/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='52'/>

Comment (GuillaumeDesforges) [inline MotorizationRate.ipynb:148]: <p>NOT THAT SIMPLE</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/Adanel/MotorizationRate/pull/3/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='55'/>

Comment (GuillaumeDesforges) [inline MotorizationRate.ipynb:148]: <p>imports at the top</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/Adanel/MotorizationRate/pull/3/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='63'/>

Comment (GuillaumeDesforges) [inline MotorizationRate.ipynb:148]: <p>MSE is not easily intepretable here</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/Adanel/MotorizationRate/pull/3/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='91'/>

Comment (GuillaumeDesforges) [inline MotorizationRate.ipynb:148]: <p>conclusion ?</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/Adanel/MotorizationRate/pull/3/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='93'/>
